### ISSUE TITLE
since 0.34, paths containing double quote, dollar, backtick, backslash break SVN::Look

### ISSUE THREAD

**Issue by stephane-chazelas** [gnustavo/SVN-Look#2]
Commit 24719cc4469aca103c91bc6d80318c195d7abd84 (meant to add support for Microsoft operating systems) introduced a security vulnerability. It changed:

```
open my $fd, '-|', @cmd, @args
```

where perl executes svnlook with its list of arguments to:

```
my $args = @args ? '"' . join('" "', @args) . '"' : '';
open my $fd, join(' ', @cmd, $args, '|');
```

That is, we build a shell command line, wrapping those _args_ in double quotes (but not for instance the repo path so spaces semicolon... in there would be a problem) and joining them with space characters before calling a shell. The shell will parse that command line and split it back into a list of arguments to pass to a svnlook command.

That's inefficient because it runs a shell for no good reason. But it's also broken because file paths are not escaped properly for the shell.

For instance, if SVN::Look is to look into a transaction that adds a file called `$(rm -rf /).txt`, it will fail and also run `rm -rf /`. because perl will run a  shell command line that looks like:

```
svnlook cmd /path/to/repo ... "path/to/$(rm -rf /).txt" ...
```

Other problematic characters would be backslash, backtick, dollar, and double-quote.

So assuming SVN::Look is used in a pre-commit hook for instance to inspect a commit transaction, that will allow users to run arbitrary commands on the svn server machine by sending specially crafted file names.

The easy fix on Unix-like OSes is to revert that  change. I don't know enough of Windows to know how to address it  there.

Raised as private bug at Ubuntu https://bugs.launchpad.net/ubuntu/+source/libsvn-look-perl/+bug/1323300

**Comment by gnustavo** [comment]
Hi, Stéphane. Thank you for the issue report.

I've reverted to the thre-arg open on non-Windows OSes. On Windows I couldn't come up with anything better than using a temporary file to intermediate the communication between the svnlook command and the Perl code.

What do you think?

**Comment by gnustavo** [comment]
I've just released version 0.40 which should fix this.
